arch/arm/slide_hash_neon.c

```diff
@@ -28,18 +30,21 @@ static inline void slide_hash_chain(Pos *table, uint32_t entries, uint16_t wsize
     Assert(sizeof(Pos) == 2, "Wrong Pos size");
     v = vdupq_n_u16(wsize);
 
-    p = (uint16x8_t *)table;
     n = size / (sizeof(uint16x8_t) * 8);
     do {
-        p[0] = vqsubq_u16(p[0], v);
-        p[1] = vqsubq_u16(p[1], v);
-        p[2] = vqsubq_u16(p[2], v);
-        p[3] = vqsubq_u16(p[3], v);
-        p[4] = vqsubq_u16(p[4], v);
-        p[5] = vqsubq_u16(p[5], v);
-        p[6] = vqsubq_u16(p[6], v);
-        p[7] = vqsubq_u16(p[7], v);
-        p += 8;
+        p0 = vld1q_u16_x4(table); 
+        p1 = vld1q_u16_x4(table+32); 
+        p0.val[0] = vqsubq_u16(p0.val[0], v);
+        p0.val[1] = vqsubq_u16(p0.val[1], v);
+        p0.val[2] = vqsubq_u16(p0.val[2], v);
+        p0.val[3] = vqsubq_u16(p0.val[3], v);
+        p1.val[0] = vqsubq_u16(p1.val[0], v);
+        p1.val[1] = vqsubq_u16(p1.val[1], v);
+        p1.val[2] = vqsubq_u16(p1.val[2], v);
+        p1.val[3] = vqsubq_u16(p1.val[3], v);
```

Comment:
This pretty much repeats four identical lines for both variables... Might be worth making it inlined function or macro even though none of the supported compilers have own inlined x4 functions/macros for anything except loads and stores...